nixos/modules/services/networking/syncthing.nix

```diff
@@ -120,9 +120,9 @@ in {
       allowedUDPPorts = [ 21027 ];
     };
 
-    systemd.packages = [ pkgs.syncthing ];
+    systemd.packages = [ pkgs.syncthing.out pkgs.syncthing.bin ];
```

Comment:
What's the problem if `pkgs.syncthing.bin` isn't included here?

Reply:
Fair question, initially it was `pkgs.syncthing` which meant `pkgs.syncthing.bin`. This was a problem, since the `bin` output does not have the systemd service files.

By using  `pkgs.syncthing.out` the reference to the binaries got lost, so I thought to include `bin` as a systemd.package.

Reply:
Looked it up, `systemd.packages` (as I thought) only uses `$out/etc/systemd` of the packages, it only adds systemd units defined in them:

https://github.com/NixOS/nixpkgs/blob/830599debecac99f0215029c922879af84be386e/nixos/modules/system/boot/systemd-lib.nix#L150-L152

So there won't be any problems with removing that (and `pkgs.syncthing.out = pkgs.syncthing`)

Reply:
> So there won't be any problems with removing that (and `pkgs.syncthing.out = pkgs.syncthing`)

I am afraid this is not correct, from my tests it is more like `pkgs.syncthing.bin = pkgs.syncthing`. 

My impression is that `buildGoPackage` in `pkgs/development/go-modules/generic/default.nix` sets

```
 outputs = args.outputs or [ "bin" "out" ];
```

so the first output ("bin") is picked when output is not defined (a.k.a. `pkgs.syncthing`).

So, we have to explicitly declare the "out" output in the `systemd.packages` and then again we are missing a reference to the other output, which contains binaries.

Reply:
Ohh, I didn't check, you're right. `pkgs.syncthing.out` in `systemd.packages` should be good then. The binaries aren't needed in `systemd.packages` though, so that can be removed.

Reply:
ok, good thing I understood `out` well.

Now, I agree that binaries are not needed in `systemd.packages`, but if I leave `syncthing.bin` out, then it is not referred anywhere else, so it is not installed. This is why I kept it.

Other approaches would be:

1. To include bin in `environment.systemPackages`.
2. To include bin in `system.extraDependencies` as mentioned in motivation.
3. To move the systemd files to $bin as suggested by @Mic92.

Personally I find 2 the most elegant, but it is mentioned that this option is primarily used by the installation tests. In the end, I will follow @Mic92's suggestion.

Reply:
In what way does putting it in `systemd.packages` make it get installed? It shouldn't. If by "install" you mean having it accessible in `$PATH`, then `environment.systemPackages` is what you want (and I suspect you do). In NixOS it doesn't make sense to "install" something just to have it be built. Either it's needed (and then you refer to it where you need it and it gets available like this), or it's not needed (so you don't refer to it and it won't be available). There's no notion of installing it just to have it installed.